syntax = "proto3";

package mongomovies;

// servi√ßo
service MongoMovies {
  rpc getMoviesById (Request) returns (Response) {}
  rpc deleteMovie (Request) returns (Response) {}
  rpc getAllMovies (Request) returns (stream Response) {}
  rpc createMovie (Request) returns (Response) {}
  rpc updateMovie (Request) returns (Response) {}
  rpc getMoviesByGenre (Request) returns (stream Response) {}
  rpc getMoviesByActor (Request) returns (stream Response) {}
}

message Genre {
  string name = 1;
}

message Cast {
  string actor = 1;
}

message Language {
  string name = 1;
}

message Director {
  string name = 1;
}

message Writer {
  string name = 1;
}

message Country {
  string name = 1;
}

message Movie {
  string id = 1;
  string plot = 2;
  repeated Genre genres = 3;
  int32 runtime = 4;
  string rated = 5;
  repeated Cast cast = 6;
  int32 num_mflix_comments = 7;
  string poster = 8;
  string title = 9;
  string fullplot = 10;
  repeated Language languages = 11;
  string released = 12;
  repeated Director directors = 13;
  repeated Writer writers = 14;
  string lastupdated = 15;
  string year = 16;
  repeated Country countries = 17;
  string type = 18;
}

message Request {
  Movie movie = 2;
  string data = 3;
}

message Response {
  bool sucess = 2;
  string message = 3;
  repeated Movie movies = 4;
}